brave/browser/renderer_host/brave_render_message_filter.cc

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "brave/browser/renderer_host/brave_render_message_filter.h"
+
+#include <stdint.h>
+
+#include <string>
+
+#include "base/bind.h"
+#include "base/bind_helpers.h"
+#include "base/logging.h"
+#include "base/macros.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/content_settings/cookie_settings_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/render_messages.h"
+#include "components/content_settings/core/browser/cookie_settings.h"
+
+using content::BrowserThread;
+
+namespace {
+
+const uint32_t kRenderFilteredMessageClasses[] = {
+    ChromeMsgStart,
+};
+
+}  // namespace
+
+BraveRenderMessageFilter::BraveRenderMessageFilter(int render_process_id,
+                                                     Profile* profile)
+    : BrowserMessageFilter(kRenderFilteredMessageClasses,
+                           arraysize(kRenderFilteredMessageClasses)),
+      render_process_id_(render_process_id),
+      profile_(profile),
+      cookie_settings_(CookieSettingsFactory::GetForProfile(profile)) {
+}
+
+BraveRenderMessageFilter::~BraveRenderMessageFilter() {
+}
+
+bool BraveRenderMessageFilter::OnMessageReceived(const IPC::Message& message) {
+  bool handled = true;
+  IPC_BEGIN_MESSAGE_MAP(BraveRenderMessageFilter, message)
+    IPC_MESSAGE_HANDLER(ChromeViewHostMsg_AllowDatabase, OnAllowDatabase)
+    IPC_MESSAGE_HANDLER(ChromeViewHostMsg_AllowDOMStorage, OnAllowDOMStorage)
+    IPC_MESSAGE_HANDLER(ChromeViewHostMsg_AllowIndexedDB, OnAllowIndexedDB)
+    IPC_MESSAGE_UNHANDLED(handled = false)
+  IPC_END_MESSAGE_MAP()
+
+  return handled;
+}
+
+void BraveRenderMessageFilter::OnAllowDatabase(
+    int render_frame_id,
+    const GURL& origin_url,
+    const GURL& top_origin_url,
+    const base::string16& name,
+    const base::string16& display_name,
+    bool* allowed) {
+  *allowed =
+      cookie_settings_->IsCookieAccessAllowed(origin_url, top_origin_url);
```

Comment:
this should be using the HostContentSettingsMap. We don't use Chrome's CookieSettingsFactory or `IsCookieAccessAllowed`. HostContentSettingsMapFactory::GetForProfile(profile_)

Reply:
same for all checks

Reply:
I can update the code - but it does work as-is. Once I fetch the HostContentSettingsMap, what would I check?

Reply:
```
 ContentSetting GetContentSetting(
      const GURL& primary_url,
      const GURL& secondary_url,
      ContentSettingsType content_type,
      const std::string& resource_identifier) const;
```